faucet/src/faucet.rs

```diff
@@ -626,30 +660,31 @@ async fn break_up_records(state: FaucetState, mut wakeup: mpsc::Receiver<()>) {
                 .await
                 .collect::<Vec<_>>();
 
-            if records.len() >= state.num_records {
-                // We have enough records again.
-                break;
+            if records.len() + 2 * transactions.len() >= state.num_records {
+                // We will have enough records again once the pending transactions finish. Return
+                // _without_ waiting for pending transactions to finish: if we know we are going to
+                // have enough records once they finish, there is no point in holding the wallet
+                // lock and just waiting. Perhaps a faucet request can be filled using the records
+                // we already have while the last few transactions are pending.
+                //
+                // Return the list of transaction receipts so the caller can wait on them if they
+                // want.
+                return Some(transactions);
             }
 
             let largest_record = match records
                 .into_iter()
                 .max_by(|x, y| x.amount().cmp(&y.amount()))
             {
-                Some(record) => record,
-                None => {
-                    tracing::warn!("No spendable records");
+                Some(record) if record.amount() >= state.grant_size * 2u64 => record,
```

Comment:
Is the check on line 613 necessary if we have to check here as well?

Reply:
The check on line 613 causes the thread to sleep if it fails. If we removed it then we would end up spamming `get_more_records` whenever we had no records to break up.